src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java

```diff
@@ -0,0 +1,272 @@
+	}
+
+	/**
+	 * @param procurementEvent
+	 * @param nonOCDS
+	 * @return suppliers as a string
+	 */
+	private String validateSuppliers(ProcurementEvent procurementEvent, MessageNonOCDS nonOCDS) {
+		// ignoring string content from organisation Ids
+		var supplierOrgIds = nonOCDS.getReceiverList().stream().map(ls -> ls.getId().replace("GB-COH-", ""))
+				.collect(Collectors.toSet());
+
+		// Retrieve and verify Tenders DB org mappings
+		var supplierOrgMappings = retryableTendersDBDelegate.findOrganisationMappingByOrganisationIdIn(supplierOrgIds);
+		if (isEmpty(supplierOrgMappings)) {
+			String errorDesc = String.format("No supplier organisation mappings found in Tenders DB %s",
+					supplierOrgIds);
+			log.error(errorDesc);
+			throw new JaggaerRPAException(errorDesc);
+		}
+
+		// Find all externalOrgIds
+		var supplierExternalIds = supplierOrgMappings.stream().map(OrganisationMapping::getExternalOrganisationId)
+				.collect(Collectors.toSet());
+
+		// Find actual suppliers of project and event
+		var suppliers = jaggaerService.getRfx(procurementEvent.getExternalEventId()).getSuppliersList().getSupplier();
+
+		// Find all unmatched org ids
+		var unMatchedSuppliers = supplierExternalIds.stream().filter(
+				orgid -> suppliers.stream().noneMatch(supplier -> supplier.getCompanyData().getId().equals(orgid)))
+				.collect(Collectors.toList());
+
+		if (!isEmpty(unMatchedSuppliers)) {
+			String errorDesc = String.format("Supplied organisation mappings not matched with actual suppliers '%s'",
+					unMatchedSuppliers);
+			log.error(errorDesc);
+			throw new JaggaerRPAException(errorDesc);
+		}
+
+		// Comparing the requested organisation ids and event suppliers info
+		var matchedSuppliers = suppliers.stream()
+				.filter(supplier -> supplierExternalIds.stream()
+						.anyMatch(orgId -> orgId.equals(supplier.getCompanyData().getId())))
+				.collect(Collectors.toList());
+
+		// Coverting all requested suppliers names into a string
+		var appendSupplierList = new StringBuilder();
+		matchedSuppliers.stream().forEach(supplier -> {
+			appendSupplierList.append(supplier.getCompanyData().getName());
+			appendSupplierList.append(";");
+		});
+
+		return appendSupplierList.toString().substring(0, appendSupplierList.toString().length() - 1);
+	}
+
+	/**
+	 * @param processInputMap
+	 * @return rpa status
+	 */
+	private String callRPAMessageAPI(Map<String, String> processInputMap) {
+		var processInput = makeProcessInput(processInputMap);
+		var request = new RPAGenericData();
+		request.setProcessInput(processInput).setProcessName(RPAProcessNameEnum.BUYER_MESSAGING.getValue())
+				.setProfileName(rpaAPIConfig.getProfileName()).setSource(rpaAPIConfig.getSource())
+				.setSourceId(rpaAPIConfig.getSourceId()).setRequestTimeout(3600000).setSync(true);
+		var response = ofNullable(rpaServiceWebClient.post().uri(rpaAPIConfig.getAccessUrl())
+				.header("Authorization", "Bearer " + getAccessToken()).bodyValue(request).retrieve()
```

Comment:
We might be able to configure this in a similar way to the main Jaggaer client if it is OAuth2 "complient" - i.e. avoid getting a new token before every request (the framework can get a token once, use it until it expires then automatically refresh it).  Perhaps one for a follow up PR 👍